automation/devops_automation_infra/utils/consul.py

```diff
@@ -63,6 +63,24 @@ def get_key(client, key):
     return res
 
 
+def get_key_layered(client, service_name, key):
+    layers_read_order = [client.OVERRIDE_KEY, client.APPLICATION_KEY, client.DEFAULT_KEY]
+    for layer in layers_read_order:
+        layered_key = f"{layer}/{service_name}/{key}"
+        value = get_key_if_exists(layered_key)
+        if value is not None:
+            return value
+    return None
+
+
+def get_key_if_exists(client, key):
+    value = None
+    res = client.kv.get(key)[1]
+    if res is not None:
+        value = res['Value']
+    return value
```

Comment:
You are keeping your functions small and clear which makes them really easy to read, Great job!
However in my opinion get_key_if_exists only makes this block of code more complex, since you are now double checking the None value and simply returning res['Value']. I would suggest you consider doing the following:

`
def get_key_layered(client, service_name, key):
    layers_read_order = [client.OVERRIDE_KEY, client.APPLICATION_KEY, client.DEFAULT_KEY]
    for layer in layers_read_order:
        layered_key = f"{layer}/{service_name}/{key}"
        value = client.kv.get(key)[1]
        if value is not None:
            return value['Value']
    return None
`

Reply:
thought about that too and if i still want to keep the exist function then changing the layered one won't really change anything or even make it less readable, Im adding example for you too look and if you feel its better that way i will keep it
`def get_key_layered(client, service_name, key):
    layers_read_order = [client.OVERRIDE_KEY, client.APPLICATION_KEY, client.DEFAULT_KEY]
    for layer in layers_read_order:
        layered_key = f"{layer}/{service_name}/{key}"
        value = client.kv.get(layered_key)[1]
        if value is not None:
            return value
    return None`